generator/generator_tests/transit_graph_test.cpp

```diff
@@ -514,26 +445,139 @@ UNIT_TEST(ClipGraph_TwoLinesTest)
+      {1 /* line id */, "1" /* number */, "Московская линия" /* title */, "subway" /* type */,
+       "green",
+       2 /* network id */, {{1, 2, 3}} /* stop id */,
+       150 /* interval */},
+      {2 /* line id */, "2" /* number */, "Варшавская линия" /* title */, "subway" /* type */,
+       "red",
+       2 /* network id */, {{5, 6}} /* stop id */,
+       150 /* interval */}
+  };
+
+  graph->m_shapes = {
       {{1, 2} /* shape id */, {{0.0, 1.0}, {2.0, 1.0}} /* polyline */},
       {{2, 3} /* shape id */, {{2.0, 1.0}, {4.0, 1.0}} /* polyline */},
       {{5, 6} /* shape id */, {{-1.0, -1.0}, {1.0, -1.0}} /* polyline */}
   };
-  TestForEquivalence(graph->GetShapes(), expectedShapes);
+
+  graph->m_networks = {
+      {2 /* network id */, "Минский метрополитен" /* title */}
+  };
+
+  return graph;
+}
+
+void TestGraph(GraphData const & graph, Graph const & expectedGraph)
+{
+  TestForEquivalence(graph.GetStops(), expectedGraph.m_stops);
+  TestForEquivalence(graph.GetGates(), expectedGraph.m_gates);
+  TestForEquivalence(graph.GetEdges(), expectedGraph.m_edges);
+  TestForEquivalence(graph.GetTransfers(), expectedGraph.m_transfers);
+  TestForEquivalence(graph.GetLines(), expectedGraph.m_lines);
+  TestForEquivalence(graph.GetShapes(), expectedGraph.m_shapes);
+  TestForEquivalence(graph.GetNetworks(), expectedGraph.m_networks);
+}
+
+void SerializeAndDeserializeGraph(GraphData const & src, GraphData & dst)
+{
+  vector<uint8_t> buffer;
+  MemWriter<decltype(buffer)> writer(buffer);
+  src.Serialize(writer);
+
+  MemReader reader(buffer.data(), buffer.size());
+  dst.DeserializeForTesting(reader);
+  TEST(dst.IsValid(), ());
+}
+
+//             ^
+//             |
+//             * 2
+//
+//   s0--------s1--------s2--------s3---s4 Line 1
+//
+//   *    *    *    *    *    *    *    *   -->
+//  -2         0              3    4    5
+//        s5--------s6 Line 2
+//
+UNIT_TEST(ClipGraph_SmokeTest)
+{
+  auto graph = CreateGraphFromJson();
+  graph->Sort();
+
+  auto expectedGraph = MakeFullGraph();
+  TestGraph(*graph, *expectedGraph);
+
+  GraphData readedGraph;
+  SerializeAndDeserializeGraph(*graph, readedGraph);
+  TestGraph(*graph, *expectedGraph);
+}
+
+//                       ^
+//                       |
+//                       * 4
+//
+//     ------------------* 3-----------Border
+//     |                                   |
+//     |                 * 2               |
+//     |                                   |
+//     |     s0----------s1----------s2----|-----s3----s4 Line 1
+//     |                                   |
+//     *-----*-----*-----*-----*-----*-----*     *     *   -->
+//    -3    -2    -1     0     1     2     3     4     5
+//                 s5----------s6 Line 2
+//
+UNIT_TEST(ClipGraph_OneLineTest)
+{
+  auto graph = CreateGraphFromJson();
+  vector<m2::PointD> points = {{3.0, 3.0}, {3.0, 0.0}, {-3.0, 0.0}, {-3.0, 3.0}, {3.0, 3.0}};
+  graph->ClipGraph({m2::RegionD(move(points))});
```

Comment:
Do we really need to move something here?

Reply:
Let's remove the `move`

Reply:
However, `points` is moved in ctor RegionD: `explicit Region(vector<PointD> && points) : m_points(move(points))`. So if I don't move here I have a compile error: `error: no matching conversion for functional-style cast from 'vector<m2::PointD>' (aka 'vector<Point<double> >') to 'm2::RegionD' (aka 'Region<Point<double> >')`

Reply:
Okay. The right way is to fix region ctor, to allow universal references instead of rvalue references.

Reply:
Yes. Done.